addon/components/o-s-s/progress-bar.ts

```diff
@@ -0,0 +1,43 @@
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+
+type ProgressBarSkins = 'primary' | 'warning' | 'success';
+
+interface OSSProgressBarArgs {
+  value: number;
+  label?: string;
+  valueIsVisible?: boolean;
+  skin?: ProgressBarSkins;
+  small?: boolean;
+}
+
+export default class OSSProgressBar extends Component<OSSProgressBarArgs> {
+  constructor(owner: unknown, args: OSSProgressBarArgs) {
+    super(owner, args);
+
+    assert('[component][OSS::ProgressBar] You must pass a numbered value between 0 and 100', args.value <= 100);
+
+    const stylesheet = document.styleSheets[0];
+    const keyframesRule = `@keyframes oss-progress-bar-animation {
+      from {
+        width: 0;
+      }
+      to {
+        width: ${this.args.value}%;
+      }
+    }`;
+    stylesheet.insertRule(keyframesRule, stylesheet.cssRules.length);
```

Comment:
Are we sure we want to do this like that ?
Can't we just define the keyframes in the less file and have it from 0 to 100 like the original upf-progress does ?

Reply:
I tried that & ended up not being able to override the 100% width attribute & had this in display : 

<img width="348" alt="Capture d’écran 2023-11-24 à 14 43 44" src="https://github.com/upfluence/oss-components/assets/37589993/ae67b629-e706-4e02-9ae4-5798a9f3528c">

Reply:
Maybe with the CSS variable you can do something. You can update the CSS variable in JS and use it in the keyframe

Reply:
+1 with handling this in w/ a CSS variable, or on the CSS side generally